.github/workflows/compare-pr-stats.yml

```diff
@@ -2,6 +2,9 @@ name: Compare PR stats
 
 on: [pull_request]
 
+permissions:
+  pull-requests: write
```

Comment:
We should have read-all at the top level and then give each job the write permissions that it needs.